state/test/test_state_proof_verification.py

```diff
@@ -75,7 +76,9 @@ def add_prefix_nodes_and_verify(state, prefix, keys_suffices=None):
     for k, v in key_vals.items():
         state.set(k, v)
 
-    prefix_prf = state.generate_state_proof_for_key_prfx(prefix.encode())
+    prefix_prf, val = state.generate_state_proof_for_keys_with_prefix(prefix.encode(), get_value=True)
+    encoded_key_values = dict(PruningState.encode_kv_for_verification(k, v) for k, v in key_vals.items())
+    assert val == encoded_key_values if len(val) == len(encoded_key_values) else val.items() >= encoded_key_values.items()
```

Comment:
What does `val.items() >= encoded_key_values.items()` assertion means here? Why can't we always assume equality?

Reply:
`encoded_key_values` is a subset of `val`. Since one test also adds a value for the prefix itself

Reply:
The condition in `add_prefix_nodes_and_verify` determines which assert needs to be called.

Reply:
Tests must be explicit and clear. In general, even loops and conditions should be avoided.
Example in this case:
- expected value (encoded_key_values) has 5 items
- val (returned by tested method) has 6 values, so we go to the `else`.
But is it expected? The tested method returned 1 more value, which is not expected in general and may be a bug which is not caught by the test.

Reply:
The helper `add_prefix_nodes_and_verify` is taking a `state` which might already have items (hence 6 values). The test is making sure that after adding the items the state has at least those elements and then doing the check.

Reply:
How can I see and detect from the test case that this is because the state already contained some items, and not because the method works incorrectly?
If the test expects 6 values as the `val` output, why don't make it check for exactly 6 values?

Reply:
Updated